.github/actions/run-python-test-set/action.yml

```diff
@@ -27,22 +27,28 @@ inputs:
     description: 'Whether to upload the performance report'
     required: false
     default: 'false'
+  artifacts-prefix:
+    description: 'Prefix to distinguish artifacts between runs'
+    required: true
+  access_key_id:
+    description: 'AWS Access Key ID'
+    required: true
+  secret_access_key:
+    description: 'AWS Secret Access Key'
+    required: true
 
 runs:
   using: "composite"
   steps:
     - name: Get Neon artifact for restoration
-      uses: actions/download-artifact@v3
+      uses: leroy-merlin-br/action-s3-cache@v1
```

Comment:
Bit of a security nitpicking, but I'd suggest when we use 3rd party actions, we should rather either specify a commit hash or then fork the action and use our own fork. Reason for that being tags can be replaced. IMHO using version tags with official ones should be fine.